### DISCUSSION TITLE
Discord.py Discussion

### DISCUSSION THREAD

**Discussion by MinhMiner672** [Rapptz/discord.py#7206] [Q&A] [answered]
Hello, I am currently using discord.py library to make the mute command. When i typed this mute command in my discord server: s.mute <user> 2h Testing (here my bot's prefix is "s." and i want to mute a user for 2 hours with the reason 'Testing'), my bot did not send any message showing that someone is just muted, it just gave the user who was muted the Muted role. Also, there was no error in the code terminal after I ran the mute command. I really dont know what i did wrong, could someone help me? I'll be very thankful.

```
@cloud.command()
@commands.has_permissions(kick_members=True)
async def mute(ctx, user:discord.Member, time_mute=str, *, reason='No reason provided'):
  muteRole = ctx.guild.get_role(861569278327390228)
  if muteRole in user.roles:
    embedAlreadyMuted = discord.Embed(title='*You are already muted*')
    await ctx.send(embed=embedAlreadyMuted)
    return

  await user.add_roles(muteRole)
  units = ['s', 'm', 'h', 'd']
  time_unit = time_mute[-1]
  if time_unit in units:
    time_dict = {'s': 1, 'm': 60, 'h': 3600, 'd':3600*24}
    value = int(time_mute[:-1])
    total_secs = value * time_dict[time_unit]
  else:
    return -1
  unit_display = ''
  if time_unit == 's':
    unit_display = 'second(s)'
  elif time_unit == 'h':
    unit_display = 'hour(s)'
  elif time_unit == 'm':
    unit_display == 'minute(s)'
  elif time_unit == 'd':
    unit_display == 'day(s)'
  
  duration_mute = discord.Embed(title='*You have been muted*', description=f'**Reason:** {reason}\n**Duration:** {value}{unit_display}', color=discord.Colour.red())

  await ctx.send(f'{user.mention}\n')
  await ctx.send(embed=duration_mute)

  asyncio.sleep(total_secs)

  await user.remove_roles(muteRole)

  embed = discord.Embed(title=f'*You have been unmuted*')
  await ctx.send(f'{user.mention}\n')
  await ctx.send(embed=embed)
```

**Comment by ImNimboss** [comment, marked as answer]
Change `asyncio.sleep(total_secs)` to `await asyncio.sleep(total_secs)`, asyncio.sleep is a [coroutine](https://docs.python.org/3/library/asyncio-task.html)

**Comment by ImNimboss** [comment]
And change `time_unit = time_mute[-1]` to `time_unit = time_mute[:-1]`

**Reply by MinhMiner672** [reply]
The bot just gave user the muted role, the embed of duration and reason was not sent

**Comment by mesub7** [comment]
```py
units = ['s', 'm', 'h', 'd']
  time_unit = time_mute[-1]
```
Will likely not work. 

return is a statement. If you use that then your function will end at that point, meaning that your bot will not send any messages. Consider using a variable to define `-1` instead.

**Reply by numbermaniac** [reply]
Note that the `return` is within an `if` block, so if the condition is false, the `return` is not reached, and the code will continue as normal.

**Comment by mikeshardmind** [comment]
You have a few things going on here which you probably want to change.

1. By muting the user before validating the time provided is valid, you break your handling of mutes. (you may want to make a converter for this)
2. Sleeping in a command to later unmute will leave users muted if the bot restarts during a mute, you probably should store temporary mutes and handle this differently.

**Comment by jelni** [comment]
Why not run this code line-by-line with a debugger and see which line stops this code? And isn't `return -1` the problem?
